lib/states/statesInMemFileDB.js

```diff
@@ -0,0 +1,576 @@
+    }
+
+    subscribe(pattern, cb) {
+        this.subscribeForClient(this.callbackSubscriptionClient, pattern, cb);
+    }
+
+    subscribeForClient(client, pattern, cb) {
+        this.handleSubscribe(client, 'state', pattern, cb);
+    }
+
+    unsubscribe(pattern, cb) {
+        this.unsubscribeForClient(this.callbackSubscriptionClient, pattern, cb);
+    }
+
+    unsubscribeForClient(client, pattern, cb) {
+        this.handleUnsubscribe(client, 'state', pattern, cb);
+    }
+
+    /**
+     * Register some instance as subscribable.
+     * If some instance says, that it is subscribable, the instance can read every time (and at start)
+     * all subscriptions to their states and will receive messages about changes of subscriptions
+     *
+     * @param instance name of instance
+     * @param cb callback which says if subscription added or yet exists
+     */
+    registerAdapterSubs(instance, cb) {
+        let added = false;
+        if (this.adapterSubs.indexOf(instance) === -1) {
+            this.adapterSubs.push(instance);
+            this.adapterSubs.sort();
+            added = true;
+        }
+        if (cb) cb(null, added);
+    }
+
+    /**
+     * Unregister instance as subscribable.
+     *
+     * @param instance name of instance
+     * @param cb callback which says if subscription removed or no
+     */
+    unregisterAdapterSubs(instance, cb) {
+        const pos = this.adapterSubs.indexOf(instance);
+        if (pos !== -1) {
+            this.adapterSubs.splice(pos, 1);
+        }
+        if (cb) cb(null, pos !== -1);
+    }
+
+    pushMessage(id, state, callback) {
+        //messagebox[id] = messagebox[id] || [];
+        state._id = this.globalMessageId++;
+        if (this.globalMessageId >= 0xFFFFFFFF) this.globalMessageId = 0;
+        //messagebox[id].unshift(state);
+        this.publishAll('messagebox', 'messagebox.' + id, state);
+        if (typeof callback === 'function') callback(null, id);
+    }
+
+    lenMessage(id, callback) {
+        if (this.messagebox[id]) {
+            if (typeof callback === 'function') callback(null, this.messagebox[id].length);
+        } else {
+            if (typeof callback === 'function') callback(tools.ERRORS.ERROR_NOT_FOUND, null);
+        }
+    }
+
+    getMessage(id, callback) {
+        if (this.messagebox[id]) {
+            if (typeof callback === 'function') callback(null, this.messagebox[id].pop());
+        } else {
+            if (typeof callback === 'function') callback(tools.ERRORS.ERROR_NOT_FOUND, null);
+        }
+    }
+
+    delMessage(id, messageId, callback) {
+        if (this.messagebox[id]) {
+            let found = false;
+            for (let i = this.messagebox[id].length - 1; i >= 0; i--) {
+                if (this.messagebox[id][i]._id === messageId) {
+                    this.messagebox[id].splice(i, 1);
+                    found = true;
+                    break;
+                }
+            }
```

Comment:
This loop is probably cleaner with `this.messagebox[id].findIndex` (see other comments above)

Reply:
solution is easier ... methods only needed as stubs for compat reason. noone writes into this.messagebox at the moment :-(
I commented out all code (but left it there)